### PR TITLE
Add support for Node.js 12.x

### PR DESCRIPTION
**Pull request by bajtos**
- Fix code to pass all tests on Node.js 12
- Add Node.js 12 to the list of Travis CI platforms

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by bajtos** [comment]
Any objections against back-porting this change to 3.x?

File: lib/model-utils.js [hunk 1]
```diff
@@ -389,6 +389,8 @@ ModelUtils._coerce = function(where, options) {
         self._coerce(clauses[k], options);
       }
 
+      where[p] = clauses;
```
Comment (bajtos) [inline lib/model-utils.js]: This is kind of weird. Before my change, the block "handle logical operators" did not apply updates made by `coerceArray` to the original `where[p]` value. As a result, array-like object `{1: {field: 'value'}}` was not converted into an array `[{field: 'value'}]`. I am not sure why this was not caught earlier and surfaced only with Node.js 12. Is it perhaps a change in how deep-equality checks work on the new Node version? IDK.

Comment (bajtos) [reply lib/model-utils.js]: Here are the tests that are failing without my fix:

https://github.com/strongloop/loopback-datasource-juggler/blob/47addd943adfd7ace37cd52d2d4dcbd04ee7c0d8/test/loopback-dl.test.js#L1498-L1503

**Comment by bajtos** [comment]
@slnode test please

**Comment by b-admike** [comment]
@slnode test please

**Pull request merged**